core/reflection.jvm/src/kotlin/reflect/jvm/internal/KCallableImpl.kt

```diff
@@ -134,6 +134,9 @@ internal abstract class KCallableImpl<out R> : KCallable<R> {
                     mask = mask or (1 shl (index % Integer.SIZE))
                     anyOptional = true
                 }
+                parameter.isVararg -> {
+                    arguments.add(parameter.type.asArrayType()?.emptyArray())
```

Comment:
In core and compiler codebase, and especially in reflection, we prefer to fail fast in any unknown scenario. So here I'd rather propose to assert that the type's JVM erasure is array first (and throw `KotlinReflectionInternalError` with a meaningful message otherwise), and then take its component type and create an empty array. And naturally this could be extracted into a single function to avoid duplication.

Reply:
Alright so for the sake of making the message more specific I'll move this logic to `KCallableImpl` since it's the one that should know what error to throw